lib/web/apiserver.go

```diff
@@ -229,6 +235,10 @@ type Config struct {
 
 	// TracerProvider generates tracers to create spans with
 	TracerProvider oteltrace.TracerProvider
+
+	// ServeAppChecker is a function that checks if the proxy can handle
+	// application requests.
+	ServeAppChecker preflightAppConnectionFunc
```

Comment:
Couldn't this field also be named `PreflightAppConnection`? The name `ServeAppChecker` is used exclusively here and I don't understand what's special about this struct that warrants a different field name. 🤔